packages/block-editor/src/components/block-mobile-toolbar/index.native.js

```diff
@@ -6,50 +6,60 @@ import { Keyboard, View } from 'react-native';
 /**
  * WordPress dependencies
  */
-import { ToolbarButton } from '@wordpress/components';
-import { __, sprintf } from '@wordpress/i18n';
 import { withDispatch, withSelect } from '@wordpress/data';
 import { compose } from '@wordpress/compose';
-import { trash } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import styles from './style.scss';
 import BlockMover from '../block-mover';
+import BlockActionsMenu from './block-actions-menu';
 import { BlockSettingsButton } from '../block-settings';
 
+const BREAKPOINTS = {
+	wrapSettings: 65,
+	wrapMover: 150,
+};
 const BlockMobileToolbar = ( {
 	clientId,
 	onDelete,
-	order,
 	isStackedHorizontally,
-} ) => (
-	<View style={ styles.toolbar }>
-		<BlockMover
-			clientIds={ [ clientId ] }
-			isStackedHorizontally={ isStackedHorizontally }
-		/>
+	blockWidth,
+	blockMobileToolbarRef,
+} ) => {
+	const wrapBlockSettings = blockWidth < BREAKPOINTS.wrapSettings;
+	const wrapBlockMover = blockWidth <= BREAKPOINTS.wrapMover;
 
-		<View style={ styles.spacer } />
+	return (
+		<View style={ styles.toolbar }>
+			{ ! wrapBlockMover && (
+				<BlockMover
+					clientIds={ [ clientId ] }
+					isStackedHorizontally={ isStackedHorizontally }
+				/>
+			) }
 
-		<BlockSettingsButton.Slot>
-			{ /* Render only one settings icon even if we have more than one fill - need for hooks with controls */ }
-			{ ( fills = [ null ] ) => fills[ 0 ] }
-		</BlockSettingsButton.Slot>
+			<View style={ styles.spacer } />
 
-		<ToolbarButton
-			title={ sprintf(
-				/* translators: accessibility text. %s: current block position (number). */
-				__( 'Remove block at row %s' ),
-				order + 1
+			{ ! wrapBlockSettings && (
+				<BlockSettingsButton.Slot>
+					{ /* Render only one settings icon even if we have more than one fill - need for hooks with controls */ }
+					{ ( fills = [ null ] ) => fills[ 0 ] }
+				</BlockSettingsButton.Slot>
 			) }
-			onClick={ onDelete }
-			icon={ trash }
-			extraProps={ { hint: __( 'Double tap to remove the block' ) } }
-		/>
-	</View>
-);
+
+			<BlockActionsMenu
```

Comment:
Do we want to render it even if we don't wrap settings and mover?

Reply:
Yes, we're going to render `BlockActionsMenu` for all the blocks. It's going to keep an option to remove a block for a _full-width_ blocks.